src/main/scala/performanceanalysis/logreceiver/alert/AlertRuleActor.scala

```diff
@@ -19,15 +19,15 @@ class AlertRuleActor(alertingRule: AlertingRule, componentId: String, metricKey:
   lazy val actionActor: ActorRef = create(context)
 
   override def receive: Receive = {
-    case msg: CheckRuleBreak if doesBreakRule(msg.value) =>
+    case CheckRuleBreak(value) if value.isInstanceOf[Duration] && doesBreakRule(value.asInstanceOf[Duration]) =>
```

Comment:
pattern matching can be improved `case CheckRuleBreak(duration: Duration) if doesBreakRule(duration)`